File: server/models/club.go
package models

import "gorm.io/gorm"

type Club struct {
	gorm.Model

	Thumb        string        `json:"thumb"`
	Image        string        `json:"image"`
	Info         string        `json:"info"`
	Name         string        `json:"name"`
	ClubAddress  *ClubAddress  `gorm:"constraint:OnDelete:CASCADE;" json:"address"`
	ClubSchedule *ClubSchedule `gorm:"constraint:OnDelete:CASCADE;" json:"schedule"`
	Equipment    []*Equipment  `gorm:"many2many:club_equipment;" json:"equipment"`
	Ticket       []Ticket      `gorm:"constraint:OnDelete:CASCADE;" json:"tickets"`
}

File: client-spa/src/shared-types/club.ts
import { IEquipment } from "./equipment";
export interface IClubSimple {
  ID: number;
  name: string;
  address: {
    street: string;
    home: string;
  };
}

export interface IClub extends IClubSimple {
  thumb: string;
  image: string;
  info: string;
  schedule: string[];
  equipment: IEquipment[];
}
